### PR TITLE
Supposed to work with Rule() method

### PR DESCRIPTION
**Pull request by DenyVeyten**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,13 +10,11 @@ module.exports = ({
...
-      root.walkRules((rule) => {
-        const safeUrlComponent = getSafeUrlComponent(rule.selector);
-        const safeUrl = utils.getSafeUrl(url, safeUrlComponent);
+    Rule(rule) {
+      const safeUrlComponent = getSafeUrlComponent(rule.selector);
+      const safeUrl = utils.getSafeUrl(url, safeUrlComponent);
 
-        rule[insertFnName]({ prop: 'background-image', value: `url('${safeUrl}')` });
-      });
+      rule[insertFnName]({ prop: 'background-image', value: `url('${safeUrl}')` });
```
Comment (jonathantneal) [inline index.js:17]:
```suggestion
      rule[insertFnName](new Declaration({ prop: 'background-image', value: `url('${safeUrl}')` }));
```

File: index.js [hunk 2]
```diff
@@ -10,13 +10,11 @@ module.exports = ({
 
   return {
     postcssPlugin: 'postcss-deadcss',
-    Root(root) {
-      root.walkRules((rule) => {
-        const safeUrlComponent = getSafeUrlComponent(rule.selector);
-        const safeUrl = utils.getSafeUrl(url, safeUrlComponent);
+    Rule(rule) {
```
Comment (jonathantneal) [inline index.js:13]:
```suggestion
    Rule(rule, { Declaration }) {
```

File: index.js (hunk 1 above)
Comment (ai) [reply index.js:17]: You do not need to wrap it to `new Declaration()`

**Comment by ai** [comment]
@DenyVeyten I do not see `postcss` in `devDependencies`. How do you install it to run locally?

Maybe PostCSS 7 used from global installation?

**Comment by DenyVeyten** [comment]
> @DenyVeyten I do not see `postcss` in `devDependencies`. How do you install it to run locally?
> 
> Maybe PostCSS 7 used from global installation?

@ai
I run `npm i --no-save postcss` before running the plugin
Let me check what version of PostCSS is run this way

**Comment by ai** [comment]
> I run npm i --no-save postcss before running the plugin

Let’s add `postcss` to `devDependencies`. This extra step with `npm i --no-save` is very rare. It will hard for other developers to send PR to your project.

**Comment by DenyVeyten** [comment]
> > I run npm i --no-save postcss before running the plugin
> 
> Let’s add `postcss` to `devDependencies`. This extra step with `npm i --no-save` is very rare. It will hard for other developers to send PR to your project.

@ai I can do like this, but what about the recommendation to leave postcss as a peerDependency? How can I achieve this recommendation?

**Comment by DenyVeyten** [comment]
@jonathantneal Thanks for the suggestions, but changes lead to the same result - it does not modify the rules

**Comment by ai** [comment]
> what about the recommendation to leave postcss as a peerDependency? How can I achieve this recommendation?

By adding `postcss` in `peerDependencies` and `devDependencies`.

You just need to avoid `dependencies`.

**Comment by DenyVeyten** [comment]
@ai Tried to install to install as dev deps, same issue :( Going to create a postcss issue
image

**Comment by ai** [comment]
OK, now it is time to start the debugger. Give me an hour.

**Comment by ai** [comment]
I found an issue in PostCSS. The fix is coming.

**Comment by ai** [comment]
The solution:

1. Use `postcss` 8.0.6
2. Add `postcss` to `devDependencies`
3. In `Rule ()` check that rule doesn’t already have `background: url(…)`.

**Pull request merged**